### PR TITLE
Allow setting throwable for CommandResult.

### PR DESCRIPTION
**Pull request by minecrafter**
**API** | [Common](https://github.com/SpongePowered/SpongeCommon/pull/503)

Resolves SpongePowered/SpongeAPI#1073

### CODE REVIEW

File: src/main/java/org/spongepowered/api/command/CommandResult.java [hunk 1]
```diff
@@ -70,20 +71,21 @@ public static CommandResult success() {
 
     /**
      * Constructs a new command result.
-     *
-     * @param successCount The success count
+     *  @param successCount The success count
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/CommandResult.java]: Oops.

File: src/main/java/org/spongepowered/api/command/CommandResult.java [hunk 2]
```diff
@@ -209,12 +222,23 @@ public Builder queryResult(@Nullable Integer queryResult) {
...
          *
          * @return A CommandResult with the specified settings
          */
         public CommandResult build() {
-            return new CommandResult(this.successCount, this.affectedBlocks, this.affectedEntities, this.affectedItems, this.queryResult);
+            return new CommandResult(this.successCount, this.affectedBlocks, this.affectedEntities, this.affectedItems, this.queryResult, this.throwable);
         }
     }
-}
+}
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/CommandResult.java]: Keep newline.

File: src/main/java/org/spongepowered/api/command/CommandResult.java [hunk 3]
```diff
@@ -70,20 +71,21 @@ public static CommandResult success() {
 
     /**
      * Constructs a new command result.
-     *
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/CommandResult.java]: Don't do this, keep the new line

File: src/main/java/org/spongepowered/api/command/CommandResult.java [hunk 4]
```diff
@@ -209,12 +222,23 @@ public Builder queryResult(@Nullable Integer queryResult) {
         }
 
         /**
+         * Sets the exception thrown from this command.
+         *
+         * @param throwable the exception thrown from this command
```
Comment (kashike) [inline src/main/java/org/spongepowered/api/command/CommandResult.java]: `@param throwable The ...` (capital `t`)

**Comment by gabizou** [comment]
@zml2008 I'd imagine this wouldn't really change much for the command refactors you have planned later on?

**Comment by zml2008** [comment]
On February 28, 2016 1:03:31 PM PST, Gabriel Harris-Rouquette <notifications@github.com> wrote:
>@zml2008 I'd imagine this wouldn't really change much for the command
>refactors you have planned later on?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/SpongePowered/SpongeAPI/pull/1075#issuecomment-189946068

While it probably wouldn't change much, it's not implemented in a way that fits at all in with the rest of the commands api, since its kinda shoved in at the highest level after all other processing occurs (so for example if another executor returned a command result with a set throwable it would be ignored. The way my work on the commands api is evolving it seems like a failure result may replace throwing exceptions, but this is probably not the way to do it.

**Comment by zml2008** [comment]
The earlier issues I have with this PR still stand and nothing else has come of it, so closing for now.

**Pull request closed without merging**